python/GafferSceneUI/ShaderUI.py

```diff
@@ -840,14 +881,36 @@ def __buttonRelease( self, pathListing, event ) :
 	def __result( self ) :
 
 		resultPaths = self.__pathListingWidget.getSelection()
+
+		if not self.__selectParameters:
+			if not len( resultPaths.paths() ):
+				return None
+
+			return resultPaths.paths()[0].strip( "/" )
+
 		resultPaths = [ self.__path.copy().setFromString( x ) for x in resultPaths.paths() ]
 
 		for p in resultPaths :
 			if len( p ) < 2 :
 				return []
-
 		return [ ( p[0], p[1] ) for p in resultPaths ]
 
 	def __updateButtonState( self, *unused ) :
 
-		self.__confirmButton.setEnabled( len( self.__result() ) > 0 )
+		result = self.__result()
+		self.__confirmButton.setEnabled( bool( result ) )
+
+		if not self.__selectParameters:
+			if result and "/" in result:
+				self.__confirmButton.setText( "Select Input Node" )
+			else:
+				self.__confirmButton.setText( "OK" )
```

Comment:
I'm inclined to second John's feedback here as I'm not overly sold on the behaviour of the confirmation button doing double-duty.

I can appreciate the potential benefits of presenting the input redirection to the user directly on the button in case they haven't learned the "click within the Input column" behaviour, but I don't think that outweighs the potential for confusion of having one button do two different things based on the selection - especially as user's first impressions of this will likely be directly after selecting a shader parameter in a very similar looking window when first selecting the parameter(s) to tweak. Having the "OK" button disable here when a shader parameter is selected and changing the wording of the window title should reinforce the concept that we're not creating a proxy from a shader parameter, but a proxy of a shader...